src/main/java/tr/edu/ogu/ceng/advice/ImsAOPStudent.java

```diff
@@ -27,6 +30,56 @@ public void beforeAddStudent(StudentDto studentDto) {
 		} else if (existingStudentStudenNo == true) {
 			throw new ServiceException("Bu öğrenci numarası zaten kayıtlı.");
 		}
-    }    
+	}
+
+	@Before("execution(* tr.edu.ogu.ceng.service.StudentService.registerAsStudent(..)) && args(request)")
+	public void beforeRegisterAsStudent(StudentDto request) {
+		validateStudentNo(request.getStudentNo());
+		validateTckno(request.getTckn());
+		checkIfStudentAlreadyRegistered(request.getStudentNo(), request.getTckn(), request.getEmail(),
+				request.getUsername());
+		checkIfPasswordsMatchingValidation(request);
+
+	}
+
+	private void checkIfStudentAlreadyRegistered(String studentNo, String tckno, String email, String username) {
+		boolean existingStudentTcNo = studentRepository.existsByTckn(tckno);
+		boolean existingStudentStudenNo = studentRepository.existsByStudentNo(studentNo);
+		boolean existingEmail = userRepository.existsByEmail(email);
+		boolean existingUsername = userRepository.existsByUsername(username);
+
+		if (existingStudentTcNo && existingStudentStudenNo) {
+			throw new ServiceException("Bu TC kimlik numarası ve Öğrenci numarası zaten kayıtlı.");
+		} else if (existingStudentTcNo) {
+			throw new ServiceException("Bu TC kimlik numarası zaten kayıtlı.");
+		} else if (existingStudentStudenNo) {
+			throw new ServiceException("Bu öğrenci numarası zaten kayıtlı.");
+		} else if (existingUsername) {
+			throw new ServiceException("Bu username zaten kayıtlı");
+		} else if (existingEmail) {
+			throw new ServiceException("Bu email zaten kayıtlı");
+		}
+
+	}
+
+	private void validateTckno(String tckno) {
+		String regex = "^[0-9]{11}$";
+		if (!tckno.matches(regex)) {
+			throw new ServiceException("TCKNO 11 haneli ve sadece sayılardan oluşmalıdır!");
+		}
+	}
+
+	private void validateStudentNo(String studentNo) {
+		String regex = "^[0-9]+$";
+		if (!studentNo.matches(regex)) {
+			throw new ServiceException("Öğrenci numarası sadece sayılardan oluşmalıdır!");
+		}
+	}
 	
+	private void checkIfPasswordsMatchingValidation(StudentDto request) {
```

Comment:
Ayrıca bu class ile aynı işlevi yapan PasswordsNotMatchedException sınıfı var orayı kullanabilirsin.

Reply:
Burada alınan iki parolanın eşleşip eşleşmediğini kontrol eden bir method aslında bu. Senin dediğin sınıfta bu işlem yapılmıyor.

Reply:
Anladım bu methoddan ayrı exception classı da eklenebilir ilerde